### PR TITLE
Rework globals init.

### PR DESCRIPTION
**Pull request by olonho**

### CODE REVIEW

File: backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/BinaryInterface.kt [hunk 1]
```diff
@@ -272,3 +272,9 @@ internal val ClassDescriptor.objectInstanceFieldSymbolName: String
 
 internal val ClassDescriptor.typeInfoHasVtableAttached: Boolean
     get() = !this.isAbstract() && !this.isExternalObjCClass()
+
+internal val String.moduleConstructorName
+    get() = "_Konan_init_${this}"
+
+internal val KonanLibraryReader.moduleConstructorName
+    get() = "_Konan_init_${uniqueName}"
```
Comment (olonho) [inline backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/BinaryInterface.kt]: Maybe better call above method.

File: backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/ContextUtils.kt [hunk 2]
```diff
@@ -344,9 +344,35 @@ internal class Llvm(val context: Context, val llvmModule: LLVMModuleRef) {
...
+    }
+
+    private fun List<LibraryReaderImpl>.topoSort(): List<LibraryReaderImpl> {
+        var sorted = mutableListOf<LibraryReaderImpl>()
+        val visited = mutableSetOf<LibraryReaderImpl>()
+        val tempMarks = mutableSetOf<LibraryReaderImpl>()
+
+        fun visit(node: LibraryReaderImpl, result: MutableList<LibraryReaderImpl>) {
+            if (visited.contains(node)) return
+            if (tempMarks.contains(node)) throw error("Cyclic dependency in library graph.")
```
Comment (olonho) [inline backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/ContextUtils.kt]: error? Maybe Error?

**Comment by olonho** [comment]
Conflicts?

File: backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/BinaryInterface.kt (hunk 1 above)
Comment (alexander-gorshenev) [reply backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/BinaryInterface.kt]: What would you propose?

Comment (olonho) [reply backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/BinaryInterface.kt]: get() =  uniqueName.moduleConstructorName

**Pull request merged**